channeld/channel.c

```diff
@@ -1091,30 +1091,42 @@ static void start_commit_timer(struct peer *peer)
 					  send_commit, peer);
 }
 
-static u8 *make_revocation_msg(const struct peer *peer, u64 revoke_index,
-			       struct pubkey *point)
+/* If old_secret is NULL, we don't care, otherwise it is filled in. */
+static void get_per_commitment_point(u64 index, struct pubkey *point,
+				     struct secret *old_secret)
 {
-	struct secret *old_commit_secret;
+	struct secret *s;
 	const u8 *msg;
 
-	/* We're revoking N-1th commit, sending N+1th point. */
 	msg = hsm_req(tmpctx,
-		      take(towire_hsm_get_per_commitment_point(NULL,
-							       revoke_index+2)));
+		      take(towire_hsm_get_per_commitment_point(NULL, index)));
 
 	if (!fromwire_hsm_get_per_commitment_point_reply(tmpctx, msg,
 							 point,
-							 &old_commit_secret))
+							 &s))
 		status_failed(STATUS_FAIL_HSM_IO,
 			      "Bad per_commitment_point reply %s",
 			      tal_hex(tmpctx, msg));
 
-	if (!old_commit_secret)
-		status_failed(STATUS_FAIL_HSM_IO,
-			      "No secret in per_commitment_point_reply %"PRIu64,
-			      revoke_index+2);
+	if (old_secret) {
```

Comment:
We might want to move this check up, and avoid the roundtrip to the HSM if we're going to discard its result anyway.